File: src/test/java/cz/cvut/fel/plichjan/distmesh/inputs/DCircleTest.java
package cz.cvut.fel.plichjan.distmesh.inputs;

import junit.framework.Assert;
import org.junit.Test;

/**
 */
public class DCircleTest {
    @Test
    public void testCall() throws Exception {
        DCircle circle = new DCircle(10, 20, 5);

        Assert.assertEquals(-5., circle.call(10, 20));
        Assert.assertEquals(0., circle.call(10, 25));
        Assert.assertEquals(0., circle.call(15, 20));
        Assert.assertEquals(5., circle.call(20, 20));
    }
}

File: src/test/kotlin/cz/cvut/fel/plichjan/distmesh/inputs/DCircleTest.kt
package cz.cvut.fel.plichjan.distmesh.inputs

import org.junit.Assert.assertEquals
import org.junit.Test

class DCircleTest {
    @Test
    @Throws(Exception::class)
    fun testCall() {
        val circle = DCircle(10.0, 20.0, 5.0)

        assertEquals(-5.0, circle.call(10.0, 20.0), 1e-9)
        assertEquals(0.0, circle.call(10.0, 25.0), 1e-9)
        assertEquals(0.0, circle.call(15.0, 20.0), 1e-9)
        assertEquals(5.0, circle.call(20.0, 20.0), 1e-9)
    }
}
